go/signedexchange/signedexchange.go

```diff
@@ -504,3 +507,40 @@ func (e *Exchange) PrettyPrintPayload(w io.Writer) {
 	fmt.Fprintf(w, "payload [%d bytes]:\n", len(e.Payload))
 	w.Write(e.Payload)
 }
+
+func (e *Exchange) JSONPrint(w io.Writer) {
+	verificationTime := time.Now()
+	certFetcher := DefaultCertFetcher
+	payload, ok := e.Verify(verificationTime, certFetcher, log.New(ioutil.Discard, "", 0))
+	// f is a copy of e except with Payload as a string, instead of a []byte (to
+	// better serialize to JSON).
```

Comment:
I'd like to avoid duplicating struct def if possible.

Would you mind omitting payload via `json:"-"` for now?

Reply:
Done (kinda—I couldn't get `json:"-"` to work quite as expected).